### PR TITLE
Advanced API Integration

### PR DESCRIPTION
**Pull request by donaldwasserman**
Apologies for the multiple PR's on this. Once I integrated this into two different apps, I discovered some bugs and needed to make some changes. 

This PR: 

- [x] Adds support for the full intercom api integration
- [x] Supports custom properties
- [x] Has tests that pass
- [x] Updated documentation

CC: @ivanvanderbyl

### CODE REVIEW

File: tests/unit/services/intercom-test.js [hunk 1]
```diff
@@ -61,11 +59,59 @@ test('it adds the correct user context to the boot config', function(assert) {
...
+ };
+ /* eslint-enable camelcase */
+
+ run(() => {
+   service.boot();
+   service.trackEvent(eventName, metadata);
+ });
+
+ assert.equal(intercomStub.calledWith('trackEvent'), true, 'Intercom track event method called');
+ sinon.assert.calledWith(intercomStub, 'trackEvent', eventName, metadata);
```
Comment (bgentry) [inline tests/unit/services/intercom-test.js]: there's some odd indentation in this file, lots of things indented by only one space

File: addon/services/intercom.js [hunk 2]
```diff
@@ -1,76 +1,370 @@
...
 
   api: intercom,
-
-  _userHashProp: computed('config.userProperties.userHashProp', function() {
+  user: null,
+  /**
+   * [description]
+   * @return {[type]} [description]
+   */
+  _userHashProp: computed('user', 'user.@each', 'config.userProperties.userHashProp', function() {
```
Comment (bgentry) [inline addon/services/intercom.js]: Isn't `@each` just for collections, not single objects?

File: addon/services/intercom.js [hunk 3]
```diff
@@ -1,76 +1,370 @@
...
+   * @private
+   * alias for appId
+   * @type {[type]}
+   */
+  appId: alias('config.appId'),
+
+
+
+  /**
+   * Boot interom window
```
Comment (bgentry) [inline addon/services/intercom.js]: typo: `interom`, plus some extra newlines above this

**Comment by mike-north** [comment]
Many of these changes were merged in #172

**Pull request closed without merging**